src/skyrimvrsavegame.cpp

```diff
@@ -2,72 +2,120 @@
 
-    QString race;
-    file.read(race); // race name (i.e. BretonRace)
+void SkyrimVRSaveGame::fetchInformationFields(
+  FileWrapper& file,
+  unsigned long& version,
+  QString& playerName,
+  unsigned short& playerLevel,
+  QString& playerLocation,
+  unsigned long& saveNumber,
+  FILETIME& creationTime) const
+{
+  unsigned long headerSize;
+  file.read(headerSize); // header size "TESV_SAVEGAME"
+  file.read(version);
+  file.read(saveNumber);
 
-    file.skip<unsigned short>(); // Player gender (0 = male)
-    file.skip<float>(2); // experience gathered, experience required
+  file.read(playerName);
 
-    FILETIME ftime;
-    file.read(ftime); //filetime
-    //A file time is a 64-bit value that represents the number of 100-nanosecond
-    //intervals that have elapsed since 12:00 A.M. January 1, 1601 Coordinated Universal Time (UTC).
-    //So we need to convert that to something useful
+  unsigned long temp;
+  file.read(temp);
+  playerLevel = static_cast<unsigned short>(temp);
 
-    //For some reason, the file time is off by about 6 hours.
-    //So we need to subtract those 6 hours from the filetime.
-    _ULARGE_INTEGER time;
-    time.LowPart = ftime.dwLowDateTime;
-    time.HighPart = ftime.dwHighDateTime;
-    time.QuadPart -= 2.16e11;
-    ftime.dwHighDateTime = time.HighPart;
-    ftime.dwLowDateTime = time.LowPart;
+  file.read(playerLocation);
 
-    SYSTEMTIME ctime;
-    ::FileTimeToSystemTime(&ftime, &ctime);
+  QString timeOfDay;
+  file.read(timeOfDay);
 
-    setCreationTime(ctime);
+  QString race;
+  file.read(race); // race name (i.e. BretonRace)
 
-    unsigned long width;
-    unsigned long height;
-    file.read(width);
-    file.read(height);
+  file.skip<unsigned short>(); // Player gender (0 = male)
+  file.skip<float>(2); // experience gathered, experience required
 
-    file.read(m_CompressionType);
+  file.read(creationTime); //filetime
+}
 
-    file.readImage(width, height, 320, true);
+std::unique_ptr<GamebryoSaveGame::DataFields> SkyrimVRSaveGame::fetchDataFields() const
+{
+  FileWrapper file(getFilepath(), "TESV_SAVEGAME"); //10bytes
 
-    file.openCompressedData();
+  unsigned long version = 0;
+  {
+    QString dummyName, dummyLocation;
+    unsigned short dummyLevel;
+    unsigned long dummySaveNumber;
+    FILETIME dummyTime;
 
-    uint8_t saveGameVersion = file.readChar();
-    uint8_t pluginInfoSize = file.readChar();
-    uint16_t other = file.readShort(); //Unknown
+    fetchInformationFields(file, version, dummyName, dummyLevel,
+      dummyLocation, dummySaveNumber, dummyTime);
+  }
 
-    file.readPlugins(1); // Just empty data
+  std::unique_ptr<DataFields> fields = std::make_unique<DataFields>();
 
-    if (saveGameVersion >= 78) {
-        file.readLightPlugins();
-    }
+  unsigned long width;
+  unsigned long height;
+  file.read(width);
+  file.read(height);
 
-    file.closeCompressedData();
-}
+  bool alpha = false;
+
+  // compatibility between LE and SE:
```

Comment:
Why is this stuff added here? was it handled in gamebryo before or something?

Reply:
Maybe it's a bad copy paste on my part. I'll check it later.